### PR TITLE
Homework

### PR DESCRIPTION
**Pull request by Raeo7**

### CODE REVIEW

File: exercises/any_odds_and_evens/main_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package main
+
+import "testing"
+
+func TestAnyIsOdd(t *testing.T) {
+	for _, tc := range []struct {
+		s    []int
+		want bool
+	}{
+		{[]int{}, false},
```
Comment (zayac) [inline exercises/any_odds_and_evens/main_test.go:10]: Please add tests with >1 elements with negative numbers.

File: exercises/any_odds_and_evens/main_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		{[]int{2}, true},
+		{[]int{3}, false},
+		{[]int{2, 4}, true},
+		{[]int{2, 1}, true},
+		{[]int{3, 7, 9}, false},
+		{[]int{2, 3, 6}, true},
+		{[]int{-2}, true},
+		{[]int{-3}, false},
+	} {
+		got := AnyIsEven(tc.s)
```
Comment (zayac) [inline exercises/any_odds_and_evens/main_test.go:44]: nit: consider writing these lines as one:
```go
if got := AnyIsEven(ts.s); got != tc.want {
  ...
}
```

Same in other programs.

File: exercises/only_odds_and_evens/only_odds_and_evens_test.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
+package main
+
+import "testing"
+
+func TestAreOddsOnly(t *testing.T) {
+	for _, tc := range []struct {
+		s    []int
+		want bool
+	}{
+		{[]int{0}, false},
```
Comment (zayac) [inline exercises/only_odds_and_evens/only_odds_and_evens_test.go:10]: What about tests containing negative, positive numbers and zeros?

File: exercises/filter_odds_and_evens/filter_odds_and_evens_test.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	}
+	return true
+}
+
+func TestFilterOdds(t *testing.T) {
+	for _, tc := range []struct {
+		s    []int
+		want []int
+	}{
+		{[]int{}, []int{}},
```
Comment (zayac) [inline exercises/filter_odds_and_evens/filter_odds_and_evens_test.go:22]: Same here. Please add a test containing positive, negative numbers and zeros.

File: exercises/rect/rect.go [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+	s := make([][]int, x)
+	for i := 0; i < x; i++ {
+		s[i] = make([]int, y)
+		for j := 0; j < y; j++ {
+			s[0][j] = 1
+			s[i][0] = 1
+			s[i][y-1] = 1
+			if i == x-1 {
+				s[i][j] = 1
+			}
```
Comment (zayac) [inline exercises/rect/rect.go:15]: This can be replaced by `s[x-1][j] = 1`.
